### PR TITLE
Added support for >>> operator and simple for loops

### PR DESCRIPTION
**Pull request by Scott-Young-6746**
#### Description

Added support for the `{VAR, NUM} >>> NUM` operator.
Added support for `for` loops of the following form:
```
for(VAR = NUM; VAR {<, <=, >=, >, ==, != } NUM; VAR = VAR {+, -, *, /} NUM;) STATEMENT
```
Where STATEMENT does not contain any lines of the form `VAR = EXPRESSION`
NUM is a constant number after the folding of constant expressions
VAR is a symbol.

#### Related Issue

Fixes issues #415  and #87 .

#### Motivation and Context

Adds language coverage for a new operator and fixes a long standing bug. 

#### How Has This Been Tested?

Tested with two new regression tests (one for each change), the `verify_odin.sh` script and a clean compile of the code submitted with issue #87 (which did not contain expected output for given input and was therefore tested only to the point were it did not reproduce the problem documented in the issue.)

#### Types of changes

- [ x ] Bug fix (change which fixes an issue)
- [ x ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ x ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ x ] I have added tests to cover my changes
- [ x ] All new and existing tests passed

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -107,3 +107,8 @@ tags
 #
 .vscode
 .history
+
+#
+# Personal circuit scratch directory
+#
+circuits/
```
Comment (kmurray) [inline .gitignore]: This looks like an individual developer configuration, it probably shouldn't end up in the master branch.

**Comment by kmurray** [comment]
Thanks for the pull request! I'll defer to @jeanlego for the ODIN related changes.

File: .gitignore (hunk 1 above)
Comment (Scott-Young-6746) [reply .gitignore]: Pushed a change to remove this from the pull request

**Comment by Scott-Young-6746** [comment]
@jeanlego I had been using tabs->spaces in my editor and noticed that the where I made changes to code it had replaced the tabs that were there with spaces. The white space changes you see in fixed indentation were replacing the 4 spaces for indentation I had used with tabs.

Assuming this is not what you are referring to in your comment, can you highlight places I missed so I can change them? I would like to ensure all my changes use the same indentation to avoid inconsistent indentation across editors.

I'll add some failing tests to the `_bug` folder and push that as soon as possible. I'll also add a test to see if creating modules in a loop works correctly.

**Pull request merged**